eq-author/src/App/QuestionnairesPage/QuestionnairesView/index.js

```diff
@@ -0,0 +1,124 @@
+import React, { useState, useRef } from "react";
+import PropTypes from "prop-types";
+import { propType } from "graphql-anywhere";
+import styled from "styled-components";
+import { isEmpty } from "lodash";
+
+import QuestionnaireSettingsModal from "App/QuestionnaireSettingsModal";
+import Button from "components/buttons/Button";
+
+import NoResults from "./NoResults";
+import QuestionnairesTable from "./QuestionnairesTable";
+import Footer from "./Footer";
+
+import reducer, { buildInitialState } from "./reducer";
+import usePersistedReducer from "./usePersistedReducer";
+
+export const STORAGE_KEY = "questionnaire-list-settings";
+const STORED_KEYS = ["currentPageIndex"];
+
+const Header = styled.div`
+  margin: 1em 0;
+  display: flex;
+  z-index: 1;
+  align-items: center;
+  justify-content: flex-end;
+`;
+
+const QuestionnairesView = ({
+  questionnaires,
+  onDeleteQuestionnaire,
+  onDuplicateQuestionnaire,
+  onCreateQuestionnaire,
+}) => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+
+  const handleModalOpen = () => setIsModalOpen(true);
+  const handleModalClose = () => setIsModalOpen(false);
+
+  const questionnairesRef = useRef(questionnaires);
+
+  const [state, dispatch] = usePersistedReducer(
+    STORAGE_KEY,
+    STORED_KEYS,
+    reducer,
+    { currentPageIndex: 0 },
+    buildInitialState(questionnairesRef.current)
+  );
+
+  if (questionnaires !== questionnairesRef.current) {
+    dispatch({
+      type: "SET_QUESTIONNAIRES",
```

Comment:
Better to declare these action types as constants? I would export from the reducer, personally, but they could live in their own file somewhere.

Reply:
Yeah, I'll move them to the reducer